wsmaster/codenvy-hosted-api-system-license/src/test/java/com/codenvy/api/license/SystemLicenseFilterTest.java

```diff
@@ -0,0 +1,239 @@
+
+        //then
+        verifyNoMoreInteractions(requestFactory);
+        verifyNoRedirectionAndError().run();
+    }
+
+    private Runnable verifySendRedirectionToAcceptFairSourcePage() {
+        return () -> {
+            try {
+                verify(servletResponse).sendRedirect(eq(ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL));
+                verify(chain, never()).doFilter(servletRequest, servletResponse);
+            } catch (Exception e) {
+                fail(e.getMessage(), e);
+            }
+        };
+    }
+
+    private Runnable verifyForbiddenError() {
+        return () -> {
+            try {
+                verify(servletResponse).setStatus(eq(HttpServletResponse.SC_FORBIDDEN));
+                verify(servletResponseWriter).write(eq(String.format("{\"message\":\"%s\"}",
+                                                                     SystemLicenseManager.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED_MESSAGE)));
+
+                verify(chain, never()).doFilter(servletRequest, servletResponse);
+            } catch (Exception e) {
+                fail(e.getMessage(), e);
+            }
+        };
+    }
+
+    private Runnable verifyNoRedirectionAndError() {
+        return () -> {
+            try {
+                verify(chain).doFilter(servletRequest, servletResponse);
+
+                verify(servletResponse, never()).sendRedirect(eq(ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL));
+
+                verify(servletResponse, never()).setStatus(eq(HttpServletResponse.SC_FORBIDDEN));
+                verify(servletResponseWriter, never()).write(eq(String.format("{\"message\":\"%s\"}",
+                                                                     SystemLicenseManager.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED_MESSAGE)));
+            } catch (Exception e) {
+                fail(e.getMessage(), e);
+            }
+        };
+    }
+
+    public void setHasFairSourceLicenseAccepted(boolean hasFairSourceLicenseAccepted) throws Exception {
+        List<IssueDto> issues;
+        if (hasFairSourceLicenseAccepted) {
+            issues = ImmutableList.of();
+        } else {
+            issues = ImmutableList.of(newDto(IssueDto.class).withStatus(Issue.Status.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED)
+                                                            .withMessage(SystemLicenseManager.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED_MESSAGE));
+        }
+
+        when(requestFactory.fromUrl(API_ENDPOINT + "/license/system/legality")).thenReturn(request);
+        when(request.useGetMethod()).thenReturn(request);
+        when(request.request()).thenReturn(response);
+        when(response.asDto(LegalityDto.class)).thenReturn(newDto(LegalityDto.class)
+                                                               .withIsLegal(false)
+                                                               .withIssues(issues));
+    }
+
+    private void setIsAdminAndNotInteractive(boolean isAdmin, boolean noUserInteraction) {
```

Comment:
It can't be used with `@BeforeTest` annotation so as it has arguments, which, in turn, should have different values for different tests.
I could simply rename this method to `init(boolean isAdmin, boolean noUserInteraction)`.